components/common/src/lib.rs

```diff
@@ -38,9 +38,14 @@ extern crate winapi;
 
 pub use self::error::{Error, Result};
 
+pub mod cli_defaults;
 pub mod command;
-pub mod defaults;
 pub mod error;
 pub mod locked_env_var;
 pub mod package_graph;
 pub mod ui;
+
+mod listen_ctl_addr;
+pub use listen_ctl_addr::ListenCtlAddr;
```

Comment:
I'm actually interested in making better and more principled use of `pub use` in the code base, particularly in parts of the manager code. And while, yes, several different types can all go into a single module, I often find this to be more troublesome to use, particularly when you get a lot of types, or the types have a lot of associated code and trait implementations ("where in this 1000 line module is the one type I care about?"). In those cases, it can actually be more useful (from an author's perspective, and in my opinion) to have a single module dedicated to all the code (and tests!) for that type (or constellation of highly related types). Then, through judicious use of `use`, `pub use`, `mod`, and `pub mod` you can create an API that's easier from a user's perspective (because having a bunch of `use listen_ctl_addr::ListenCtlAddr;` lines is admittedly pretty rough).

In this particular case, I agree that gathering things together under a higher-level `types` namespace could provide some better indication of sourcing and logical grouping; `use common::types::ListenCtlAddr;` seems very readable and obvious to me. However, that doesn't necessarily mean that it'd be better to place everything into a single `types.rs`. 

As with all things, moderation is key here, but I think it could be a pretty powerful tool, and could drive some more principled visibility decisions.

Reply:
Is there some standard practice around this? I agree with @christophermaier here, but I'm happy to go with whatever pattern we decide to use.

Reply:
My priorities are:
1. Don't fall prey to analysis paralysis and spend too much time on this; there's no perfect solution, but time, experience and refactoring will get us closer
1. Prefer putting things that are logically related in the same module over keeping files small
1. Avoid redundancy like `use foo::Foo;` (see "[Name Things Once](https://www.youtube.com/watch?v=1__lNTlj1_w)")
1. Avoid confusing statements like `use Foo;` instead of `use bar::Foo` caused by `pub use`
1. Use `pub use` so consumers can write `use cratename::Type;` instead of `use cratename::internal::implementation::specific::details::Type;`

Reply:
We've not really had any principled discussions around this, so this is the beginning. For sure don't spend a lot of time sweating over this; time will show us what works best.

@baumanj's guidelines are a great start, though.